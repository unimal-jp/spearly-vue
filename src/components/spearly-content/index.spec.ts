import { mount, VueWrapper } from '@vue/test-utils'
import SpearlyContent from './index.vue'

describe('SpearlyContent', () => {
  let wrapper: VueWrapper
  const getContentMock = jest.fn().mockResolvedValue({
    attributes: {
      patternName: 'a',
    },
    id: '1',
  })
  const getContentPreviewMock = jest.fn().mockResolvedValue({})
  const pageViewMock = jest.fn().mockResolvedValue({})

  beforeEach(() => {
    wrapper = mount(SpearlyContent, {
      props: {
        id: 'CONTENT_ID',
      },
      global: {
        provide: {
          $spearly: {
            getContent: getContentMock,
            getContentPreview: getContentPreviewMock,
          },
          $spearlyAnalytics: {
            pageView: pageViewMock,
          },
        },
      },
    })
  })

  afterEach(() => {
    getContentMock.mockClear()
    getContentPreviewMock.mockClear()
  })

  describe('initialized', () => {
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('get content', () => {
    it('call getContent if preview-token prop is not specified', () => {
      expect(getContentMock).toHaveBeenCalledWith('CONTENT_ID', {})
    })

    it('call getContentPreview if preview-token prop is specified', () => {
      wrapper = mount(SpearlyContent, {
        props: {
          id: 'CONTENT_ID',
          previewToken: 'TOKEN',
        },
        global: {
          provide: {
            $spearly: {
              getContent: getContentMock,
              getContentPreview: getContentPreviewMock,
            },
            $spearlyAnalytics: {
              pageView: pageViewMock,
            },
          },
        },
      })
      expect(getContentPreviewMock).toHaveBeenCalledWith('CONTENT_ID', 'TOKEN')
    })
  })
})
