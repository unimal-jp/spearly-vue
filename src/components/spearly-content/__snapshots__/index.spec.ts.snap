// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpearlyContent initialized snapshot 1`] = `
<div>
  
  <!--v-if-->
  
</div>
`;
